ksql-functional-tests/src/main/java/io/confluent/ksql/test/serde/avro/ValueSpecAvroSerdeSupplier.java

```diff
@@ -126,6 +125,17 @@ private static Object valueSpecToAvro(final Object spec, final org.apache.avro.S
         case BOOLEAN:
           return spec;
         case ARRAY:
+          if (schema.getElementType().getName().equals(AvroData.MAP_ENTRY_TYPE_NAME)
+              ||
+              Objects.equals(
+                  schema.getElementType().getProp(AvroData.CONNECT_INTERNAL_TYPE_NAME),
+                  AvroData.MAP_ENTRY_TYPE_NAME)
+              ) {
+            return ((Map<Object, Object>) spec).entrySet().stream()
+                .map(objectObjectEntry ->
+                    getAvroRecordForMapEntry(objectObjectEntry, schema.getElementType()))
+                .collect(Collectors.toList());
```

Comment:
Yes, this is confusing since this is a special case that connect handles Maps. Since in avro format map keys can only be string, connect represents the maps as an array of entries so that they can handle non string map keys. SO when we have a map to serialize we serialize it as an array of map entries and when we have an array of map entries to deserialize we deserialize it as a map!
I'll add more comments to the code to clarify this.